src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -86,7 +86,7 @@ public String send(Letter letter, String serviceName) throws JsonProcessingExcep
             throw new SendMessageException("Could not send message to ServiceBus with " + messageId);
         }
 
-        return messageId;
+        return id.toString();
```

Comment:
Not strictly related with this pull request, but I think we have a bigger problem here - we don't wait for the queue client to send the message - we'll return the ID and store the record regardless of whether it succeeded or failed.

Reply:
ah true, forgot to investigate the whole block.

wasn't the decision to save to the db after successful publish to the queue? could attach another async future call in that block to do so

Reply:
We're storing the record in the DB before sending the message to the queue. I suppose the idea is that if we fail somewhere between there and the end of the method, Spring will rollback the DB transaction.